tegra/tegra.c

```diff
@@ -189,12 +189,12 @@ int drm_tegra_bo_wrap(struct drm_tegra_bo **bop, struct drm_tegra *drm,
 	return 0;
 }
 
-struct drm_tegra_bo *drm_tegra_bo_ref(struct drm_tegra_bo *bo)
+int drm_tegra_bo_ref(struct drm_tegra_bo *bo)
 {
-	if (bo)
-		atomic_inc(&bo->ref);
+	if (!bo)
+		return -EINVAL;
 
-	return bo;
+	return atomic_inc_return(&bo->ref);
```

Comment:
since we're returning either (negative) errors or refcounts here, maybe we should verify that the returned refcount isn't negative as well (due to API misuse)? An assert is probably enough...